doc/release/1.16.0-notes.rst

```diff
@@ -106,6 +106,13 @@ details.
 The ``out`` argument to these functions is now always tested for memory overlap
 to avoid corrupted results when memory overlap occurs.
 
+``np.reshape`` and the ``reshape`` allow to force no-copy mode
+--------------------------------------------------------------
+The reshape function in numpy will normally return a view but create a new
+array if necessary. A new ``copy`` keyword argument can toggle this behaviour.
+``copy=True`` will always copy the data, while ``copy=False`` will raise an
```

Comment:
This is not consistent with the meaning of `copy=False` elsewhere. Right now, we have:

* `copy = True` - always copy
* `copy = False` - copy only if necessary

I think we should indicate never copy with `"never"`, `None`, or `"raise"`

Reply:
Yes, I suppose we should probably handle the `copy` argument consistently with `np.array()`, for which `copy=False` means "only copy if necessary".

I like `copy='never'`

Reply:
So, I could just implement that on the C-Side (and also for reshape). But, for all exposed python functions that currently have the flag (mostly `np.asarray`, but `.astype` probably as well), the only way that does not require at least 2 years of waiting would be to create a `np.nevercopy` singleton, which would have to be supported indefinitely…

Reply:
Or, we do a funny middle way: Actually hard deprecate/remove the specific "never" string (who would use it?! :)) now. Just to speed things up. Should still have at least one version where "never" will simply break though.

Reply:
So `bool(None) is False` which is really strange currently, because it does not give the default of course. Anyway, doesn't matter. The only possible thing would be to ignore it for all but `reshape` or create a singleton and deprecate as much as possible